Coding/server/lib/Store/Store.cpp

```diff
@@ -104,53 +104,67 @@ void Store::setKeyCert(const KeyCert& keycert)
 
 bool Store::saveKeyCert()
 {
-    uint8_t* keyBuffer = new uint8_t[KeyCert::RSA_KEY_SIZE_BYTE];
-    uint8_t* certBuffer = new uint8_t[KeyCert::CERT_SIZE_BYTE];
-
     /* Stores if writing to NVS was successful */
     bool putRSAResult = false;
     bool putCertResult = false;
-
     bool retCode = false;
 
-    m_keyCert.serialize(keyBuffer, certBuffer);
-    putRSAResult = m_nvsmgr.putEntry("rsakey", keyBuffer, KeyCert::RSA_KEY_SIZE_BYTE);
-    putCertResult = m_nvsmgr.putEntry("sslcert", certBuffer, KeyCert::CERT_SIZE_BYTE);
-
-    delete[] keyBuffer;
-    delete[] certBuffer;
+    uint8_t* keyBuffer = new uint8_t[KeyCert::RSA_KEY_SIZE_BYTE];
+    uint8_t* certBuffer = new uint8_t[KeyCert::CERT_SIZE_BYTE];
 
-    retCode = (true == putRSAResult) && (true == putCertResult);
-    if (false == retCode)
+    if ((nullptr != keyBuffer) && (nullptr != certBuffer))
     {
-        LOG_ERROR("Could not save KeyCert to persistent storage");
+        m_keyCert.serialize(keyBuffer, certBuffer);
+        putRSAResult = m_nvsmgr.putEntry("rsakey", keyBuffer, KeyCert::RSA_KEY_SIZE_BYTE);
+        putCertResult = m_nvsmgr.putEntry("sslcert", certBuffer, KeyCert::CERT_SIZE_BYTE);
+
+        delete[] keyBuffer;
+        delete[] certBuffer;
+
+        retCode = (true == putRSAResult) && (true == putCertResult);
+        if (false == retCode)
+        {
+            LOG_ERROR("Could not save KeyCert to persistent storage");
+        }
+    }
+    else
+    {
+        LOG_ERROR("Out of memory. Could not create buffers for KeyCert serialization!");
```

Comment:
Memory leak in case one of them got the memory and the other not.

Reply:
Fixed with 88c3418